kratos/includes/element.h

```diff
@@ -981,6 +981,8 @@ class Element : public GeometricalObject, public Flags
                                             Matrix& rOutput,
                                             const ProcessInfo& rCurrentProcessInfo)
     {
+         if (rOutput.size1() != 0)
+	        rOutput.resize(0, 0, false);
```

Comment:
Do you need this resize? Is it for distinguish it in the output?

Reply:
We do a check for the size outside, and if it is zero, we know there is no contribution and we skip the further computation of the sensitivity related to this element.

As an alternative is should work to give it a proper size, and then just fill it with Zeros to further without a check for the size outside, e.g:
```cpp
int number_of_design_variables = 1; //1 for the scalar version, 3 for the vector version of this function
if ((rOutput.size1() != number_of_design_variables) or (rOutput.size2() != number_of_dofs)
     rOutput.resize(number_of_design_variables, number_of_dofs, false);
noalias( rOutput) = ZeroMatrix( number_of_design_variables, number_of_dofs)
```

Reply:
I would say that the resize to zero as you did  is more efficient

Reply:
@pooyan-dadvand we have the same problem in the fluid adjoints, and we have independently reached the same solution as @armingeiser here. I'd say that having the size zero output is very convenient to check for do-nothing cases in the calling function.

Reply:
:+1: